drivers/adc-dac/ad5592r/ad5592r-base.c

```diff
@@ -320,3 +320,184 @@ int32_t ad5592r_reset_channel_modes(struct ad5592r_dev *dev)
+		return -1;
+
+	ret = ad5592r_base_reg_read(dev, reg_addr, &temp_reg_val);
+	if (ret < 0)
+		return ret;
+
+	temp_reg_val &= ~mask;
+	temp_reg_val |= data;
+
+	ret = ad5592r_base_reg_write(dev, reg_addr, temp_reg_val);
+	if (ret < 0)
+		return ret;
+
+	return 0;
+}
+
+/**
+ * Set ADC Range of the device
+ *
+ * @param dev - The device structure.
+ * @param adc_range - ADC Range
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_set_adc_range(struct ad5592r_dev *dev,
+			      enum ad559xr_range adc_range)
+{
+	int32_t ret;
+	uint16_t status;
+
+	if (!dev)
+		return -1;
+
+	if (adc_range)
+		status |= AD5592R_REG_CTRL_ADC_RANGE;
+	else
+		status &= ~AD5592R_REG_CTRL_ADC_RANGE;
+
+	ret = ad5592r_base_reg_update(dev, AD5592R_REG_CTRL, status,
+				      AD5592R_REG_CTRL_ADC_RANGE);
+	if (ret < 0)
+		return ret;
+
+	dev->adc_range = adc_range;
+
+	return 0;
+}
+
+/**
+ * Set DAC Range of the device
+ *
+ * @param dev - The device structure.
+ * @param dac_range - DAC Range
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_set_dac_range(struct ad5592r_dev *dev,
+			      enum ad559xr_range dac_range)
+{
+	int32_t ret;
+	uint16_t status;
+
+	if (!dev)
+		return -1;
+
+	if (dac_range)
+		status |= AD5592R_REG_CTRL_DAC_RANGE;
+	else
+		status &= ~AD5592R_REG_CTRL_DAC_RANGE;
+
+	ret = ad5592r_base_reg_update(dev, AD5592R_REG_CTRL, status,
+				      AD5592R_REG_CTRL_DAC_RANGE);
+	if (ret < 0)
+		return ret;
+
+	dev->dac_range = dac_range;
+
+	return 0;
+}
+
+/**
+ * Set Power Down DAC Channel of the device
+ *
+ * @param dev - The device structure.
+ * @param chan - The channel number.
+ * @param status - Status to enable/disable power down.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_power_down(struct ad5592r_dev *dev, uint8_t chan, bool status)
+{
+	int ret;
+	uint16_t temp_reg_val;
+
+	if (!dev)
+		return -1;
+
+	if (status)
+		temp_reg_val |= NO_OS_BIT(chan);
+	else
+		temp_reg_val &= ~NO_OS_BIT(chan);
+
+	ret = ad5592r_base_reg_update(dev, AD5592R_REG_PD, temp_reg_val,
+				      NO_OS_BIT(chan));
+	if (ret < 0)
+		return ret;
+
+	dev->power_down[chan] = status;
+
+	return 0;
+}
+
+/**
+ * Set Reference Select option for the device
+ *
+ * @param dev - The device structure.
+ * @param status - Status to enable/disable internal reference.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_set_int_ref(struct ad5592r_dev *dev, bool status)
+{
+	uint16_t temp_reg_val;
+	int ret;
+
+	if (status)
+		temp_reg_val |= AD5592R_REG_PD_EN_REF;
+	else
+		temp_reg_val &= ~AD5592R_REG_PD_EN_REF;
+
+	ret = ad5592r_base_reg_update(dev, AD5592R_REG_PD, temp_reg_val,
+				      AD5592R_REG_PD_EN_REF);
+	if (ret < 0)
+		return ret;
+
+	dev->int_ref = status;
+
+	return 0;
+}
+
+/**
+ * Set ADC Buffer for the device
+ *
+ * @param dev - The device structure.
+ * @param status - Status to enable/disable adc buffer.
```

Comment:
i wouldn't name it status, rename it to `enable`, then it's even more clear what it does